skypy/halos/tests/test_colossus.py

```diff
@@ -32,3 +32,76 @@ def test_colossus_mass_sampler():
     D, p = kstest(sample, lambda t: np.interp(t, m, CDF))
 
     assert p > 0.01, 'D = {}, p = {}'.format(D, p)
+
+
+@pytest.mark.skipif(not HAS_COLOSSUS, reason='test requires colossus')
+@pytest.mark.flaky
+def test_colossus_mf_redshift():
+
+    from skypy.halos.redshift import colossus_mf_redshift
+    from astropy.cosmology import Planck18
+    from astropy import units
+    from scipy import integrate
+    from colossus.lss.mass_function import massFunction
+
+    # Parameters
+    redshift = np.linspace(0., 2., 100)
+    model, mdef = 'despali16', '500c'
+    m_min, m_max = 1e+12, 1e+16
+    sky_area = 1.0 * units.deg**2
+    cosmology = Planck18
+    sigma8, ns = 0.82, 0.96
+
+    # Sample redshifts
+    z_halo = colossus_mf_redshift(redshift, model, mdef, m_min, m_max, sky_area,
+                                  cosmology, sigma8, ns, noise=False)
+
+    # Integrate the mass function to get the number density of halos at each redshift
+    def dndlnM(lnm, z):
+        return massFunction(np.exp(lnm), z, 'M', 'dndlnM', mdef, model)
+    lnmmin = np.log(m_min/cosmology.h)
+    lnmmax = np.log(m_max/cosmology.h)
+    density = [integrate.quad(dndlnM, lnmmin, lnmmax, args=(z))[0] for z in redshift]
+    density = np.array(density) * np.power(cosmology.h, 3)
+
+    # Integrate total number of halos for the given area of sky
+    density *= (sky_area * cosmology.differential_comoving_volume(redshift)).to_value('Mpc3')
+    n_halo = np.trapz(density, redshift, axis=-1)
+
+    # make sure noise-free sample has right size
+    assert np.isclose(len(z_halo), n_halo, atol=1.0)
+
+    # Halo redshift CDF
+    cdf = density  # same memory
+    np.cumsum((density[1:]+density[:-1])/2*np.diff(redshift), out=cdf[1:])
+    cdf[0] = 0
+    cdf /= cdf[-1]
+
+    # Check distribution of sampled halo redshifts
+    D, p = kstest(z_halo, lambda z_: np.interp(z_, redshift, cdf))
+    assert p > 0.01, 'D = {}, p = {}'.format(D, p)
+
+
+@pytest.mark.skipif(not HAS_COLOSSUS, reason='test requires colossus')
+def test_colossus_mf():
+
+    from skypy.halos import colossus_mf
+    from astropy.cosmology import Planck18
+    from astropy import units
+
+    # redshift and mass distributions are tested separately
+    # only test that output is consistent here
+
+    z = np.linspace(0., 1., 100)
+    model, mdef = 'despali16', '500c'
+    m_min, m_max = 1e+12, 1e+16
+    sky_area = 1.0 * units.deg**2
+    cosmo = Planck18
+    sigma8, ns = 0.82, 0.96
+    z_halo, m_halo = colossus_mf(z, model, mdef, m_min, m_max, sky_area, cosmo, sigma8, ns)
```

Comment:
Same Warning here `WARNING: Astropy cosmology class contains massive neutrinos, which are not taken into account in Colossus.`